### PR TITLE
Fix documentation on DataPack methods

### PR DESCRIPTION
**Pull request by Deathreus**
Also fixup a couple descriptions in the methodmap

### CODE REVIEW

File: plugins/include/datapack.inc [hunk 1]
```diff
@@ -58,6 +58,16 @@ methodmap DataPack < Handle
...
 	public native void WriteFloat(float val, bool insert = false);
+	
+	// Packs a vector into a data pack.
+	//
+	// @param val		Float array to add.
+	// @param insert	Determines whether mid-pack writes will insert instead of overwrite.
+	public void WriteVector(const float val[3], bool insert = false)
+	{
+		for(int x=0; x<3; ++x)
+			this.WriteFloat(val[x]);
```
Comment (rumblefrog) [inline plugins/include/datapack.inc]: Seems like you forgot to pass insert to WriteFloat

File: plugins/include/datapack.inc [hunk 2]
```diff
@@ -135,6 +154,19 @@ native void WritePackCell(Handle pack, any cell);
...
 native void WritePackFloat(Handle pack, float val);
 
+/**
+ * Packs a vector into a data pack.
+ *
+ * @param pack		Handle to the data pack.
+ * @param val			Float array to add.
+ * @error			Invalid handle.
+ */
+stock void WritePackVector(Handle pack, const float val[3])
```
Comment (Headline) [inline plugins/include/datapack.inc]: We don't need to add oldstyle decl for new features

File: plugins/include/datapack.inc [hunk 3]
```diff
@@ -171,6 +203,19 @@ native any ReadPackCell(Handle pack);
...
 native float ReadPackFloat(Handle pack);
 
+/**
+ * Reads a vector from a data pack.
+ *
+ * @param pack		Handle to the data pack.
+ * @param buffer		Destination array.
+ * @error			Invalid handle, or bounds error.
+ */
+stock void ReadPackVector(Handle pack, float buffer[3])
```
Comment (Headline) [inline plugins/include/datapack.inc]: ditto

**Comment by FortyTwoFortyTwo** [comment]
Would it be better to have `WriteCellArray`, `WriteFloatArray`, `ReadCellArray` and `ReadFloatArray` instead? Allowing to pass any array size rather than limited to 3, write/read vectors can be done using `WriteFloatArray`/`ReadFloatArray` with size 3.

**Comment by Deathreus** [comment]
> Would it be better to have `WriteCellArray`, `WriteFloatArray`, `ReadCellArray` and `ReadFloatArray` instead? Allowing to pass any array size rather than limited to 3, write/read vectors can be done using `WriteFloatArray`/`ReadFloatArray` with size 3.

Actually, yeah, that would be far better, I'll commit that instead at the native level

**Comment by Headline** [comment]
After conversation with @Deathreus we'll be leaving the vector changes for later and take the documentation ones for now.

**Comment by peace-maker** [comment]
What happened with the submodules? Revert those changes before merging!

**Comment by Headline** [comment]
@peace-maker thank you for keeping an eye out, but I’m aware.

**Pull request merged**